gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/utils/gcp/gcp_secret_manager_client.py

```diff
@@ -0,0 +1,102 @@
+#   Copyright (C) 2023-2024 Credit Mutuel Arkea
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""API Client to consume GCP Secret Manager API"""
+
+import json
+import logging
+from typing import Optional, Type, TypeVar
+
+from google.api_core.exceptions import NotFound
+from google.cloud import secretmanager
+
+from gen_ai_orchestrator.configurations.environment.settings import application_settings
+from gen_ai_orchestrator.models.security.ai_provider_secret import AIProviderSecret
+from gen_ai_orchestrator.models.security.credentials import Credentials
+
+logger = logging.getLogger(__name__)
+T = TypeVar('T')
+
+
+class GCPSecretManagerClient:
```

Comment:
Using python abc package you could create an abstract class (that act as an interface), and inherit it for the AWS and GCP Secret Manager, this will ensure that they share the same contract / methods (otherwise it would raise exception if the method isn't implemented).

Reply:
absolutely

Reply:
I added a new comment on the changed version more explicit than this one, sorry for the misunderstanding